e2etest/e2etest_suite_test.go

```diff
@@ -70,3 +83,218 @@ func TestE2E(t *testing.T) {
+func getL2AddressPoolRange() ([]string, []string, error) {
+	// Get the used ipv4 and ipv6 addresses.
+	networkSubnet, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .IPAM.Config}}{{.Subnet}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get subnet network info: %s", err)
+	}
+	networkSubnet = strings.TrimSpace(networkSubnet)
+
+	usedIpv4, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .Containers}}{{.IPv4Address}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get ipv4 addresses: %s", err)
+	}
+	usedIpv4 = strings.TrimSpace(usedIpv4)
+
+	usedIpv6, err := runCommand(false, "docker", "network", "inspect", "-f", "{{range .Containers}}{{.IPv6Address}} {{end}}", "kind")
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get ipv6 addresses: %s", err)
+	}
+	usedIpv6 = strings.TrimSpace(usedIpv6)
+
+	// Get the address pool range.
+	ipv4Range := make([]string, 2)
+	ipv6Range := make([]string, 2)
+	for _, network := range strings.Split(networkSubnet, " ") {
+		// Check is IPv4 network
+		if strings.Count(network, ":") < 2 {
+			// Sort the ips to get the last used ip address.
+			usedIpv4List := sortIPs(strings.Split(usedIpv4, " "))
+			lastusedIP := usedIpv4List[len(usedIpv4List)-1]
+			nextIP, err := incrementIPByValue(lastusedIP, network, 1)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get first free ipv4 address: %s", err)
+			}
+			// Get a range of 10 unused ipv4 addresses.
+			ipv4Range, err = getIPsRange(nextIP, network, 10)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get ipv4 range: %s", err)
+			}
+		} else {
+			// Sort the ips to get the last used ip address.
+			usedIpv6List := sortIPs(strings.Split(usedIpv6, " "))
+			lastusedIP := usedIpv6List[len(usedIpv6List)-1]
+			nextIP, err := incrementIPByValue(lastusedIP, network, 1)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get first free ipv6 address: %s", err)
+			}
+			// Get a range of 10 unused ipv6 addresses.
+			ipv6Range, err = getIPsRange(nextIP, network, 10)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Failed to get ipv6 range: %s", err)
+			}
+		}
+	}
+
+	return ipv4Range, ipv6Range, nil
+}
+
+// Configure MetalLB for testing.
+// Identify the unused network address range and use it in configmap.
+func configureMetalLB(cs clientset.Interface) error {
+	err := configureFRR(cs)
+	if err != nil {
+		return fmt.Errorf("Failed to configure FRR: %s", err)
+	}
+
+	// TODO: should set IPv6 addresspool as well when adding support for IPv6.
+	ipv4Range, _, err := getL2AddressPoolRange()
+	if err != nil {
+		return fmt.Errorf("Failed to get layer2 address pool range: %s", err)
+	}
+
+	peerAddress, err := runCommand(false, "docker", "inspect", "-f", "{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}", consts.FRRContainerName)
```

Comment:
Can you foresee a test case in which we have two different peers? i.e. two FRR Containers?